src/render/src/macros/mod.rs

```diff
@@ -29,3 +29,56 @@ macro_rules! gfx_format {
         }
     }
 }
+
+
+/// Helps define vertex, constant and pipeline formats
+/// # Example
+/// ```
+/// gfx_defines! {
```

Comment:
Do you mind having a simple examples in the "tests/macros.rs" rather then here? Gotta make sure the macro spews correct code.